pkgs/development/python-modules/weasyprint/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib, isPy3k, fetchPypi, buildPythonPackage
+, glib, pango, fontconfig
+, cffi, pyphen, cairosvg, html5lib, tinycss2, cairocffi, cssselect2
+, pytestcov, pytestrunner, pytest-flake8, pytest-isort }:
+
+buildPythonPackage rec {
+  pname = "weasyprint";
+  version = "44";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "WeasyPrint";
+    sha256 = "0f5bzwac1c4wgsscdv6d2d08n6g3j7pmj76ja5q31v40vzkz96vr";
+  };
+
+  patches = [
+    ./ld-library-path.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace weasyprint/fonts.py \
+      --subst-var-by libpangoft2-1-0 ${lib.makeLibraryPath [ pango ]}/libpangoft2-1.0.so \
+      --subst-var-by libfontconfig ${lib.makeLibraryPath [ fontconfig ]}/libfontconfig.so.1
+
+    substituteInPlace weasyprint/text.py \
+      --subst-var-by libgobject-2-0 ${lib.makeLibraryPath [ glib ]}/libgobject-2.0.so \
+      --subst-var-by libpango-1-0 ${lib.makeLibraryPath [ pango ]}/libpango-1.0.so \
+      --subst-var-by libpangocairo-1-0 ${lib.makeLibraryPath [ pango ]}/libpangocairo-1.0.so
```

Comment:
Why not use `substituteAll`? That's at least what I usually do but it's up to personal preference.

Reply:
No, instead of using `substituteInPlace` you can use `substituteAll` inside `patches`. Just search for it and you'll see how it works.
Btw, you can e.g. put `libgobject-2-0 = "${lib.makeLibraryPath [ glib ]}/libgobject-2.0.${stdenv.hostPlatform.extensions.sharedLibrary}"` in there.

Reply:
Oh, I only knew about the `substituteAll` as a function `substituteAll inFile outFile` from nixpkgs/manual, and turns out there is a equivalent of `makeLibraryPath` for one package, `getLib`, so I'll change that to.